sql/src/test/java/io/crate/expression/symbol/AggregationTest.java

```diff
@@ -51,4 +52,24 @@ public void test_serialization_with_filter() throws Exception {
         assertThat(expected.filter(), is(Literal.BOOLEAN_FALSE));
         assertThat(expected, is(actual));
     }
+
+    @Test
+    public void test_serialization_with_filter_before_version_4_1_0() throws Exception {
+        Aggregation actual = new Aggregation(
+            CountAggregation.COUNT_STAR_FUNCTION,
+            CountAggregation.COUNT_STAR_FUNCTION.returnType(),
+            List.of()
+        );
+        BytesStreamOutput output = new BytesStreamOutput();
+        output.setVersion(Version.V_4_0_0);
+        Symbols.toStream(actual, output);
+
+        StreamInput input = output.bytes().streamInput();
+        input.setVersion(Version.V_4_0_0);
```

Comment:
is this necessary as afaik the output defined the version already?

Reply:
it is necessary, coz `Symbols.toStream` just calls `writeTo` of `Aggregation`, which does not write the version into the stream output. Seems that only stream input/outputs do that at instantization.

Reply:
i merged this pr without changing this particular test and probably will go for merging the feature PR such as it is approved already. I'd say it is not that critical, and if there is something that I am really missing here, i can follow up with another PR.